webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DatasetDeleter.scala

```diff
@@ -31,23 +39,244 @@ trait DatasetDeleter extends LazyLogging with DirectoryConstants {
         case e: Exception                                => Fox.failure(s"Deleting dataset failed: ${e.toString}", Full(e))
       }
 
+    def moveToTrash(organizationId: String,
+                    datasetName: String,
+                    dataSourcePath: Path,
+                    reason: Option[String]): Fox[Unit] =
+      if (Files.exists(dataSourcePath)) {
+        val trashPath: Path = dataBaseDir.resolve(organizationId).resolve(trashDir)
+        val targetPath = trashPath.resolve(datasetName)
+        new File(trashPath.toString).mkdirs()
+
+        logger.info(s"Deleting dataset by moving it from $dataSourcePath to $targetPath${if (reason.isDefined)
+          s" because ${reason.getOrElse("")}"
+        else "..."}")
```

Comment:
```suggestion
        logger.info(s"Deleting dataset by moving it from $dataSourcePath to $targetPath${reason.map(r => s"because $r").getOrElse("...")}")
```

Reply:
I don't think I wrote this but happy to change  :smile: